File: src/ros2_bridge/mappers/foxglove_msgs/point3.rs
//! Typed mapper for `foxglove_msgs/msg/Point3`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn point3_to_bus(msg: ros_env::foxglove_msgs::msg::Point3) -> crate::foxglove_msgs::msg::v1::Point3 {
    crate::foxglove_msgs::msg::v1::Point3 {
        x: msg.x,
        y: msg.y,
        z: msg.z,
    }
}

pub(crate) fn point3_to_ros(bus: crate::foxglove_msgs::msg::v1::Point3) -> ros_env::foxglove_msgs::msg::Point3 {
    ros_env::foxglove_msgs::msg::Point3 {
        x: bus.x,
        y: bus.y,
        z: bus.z,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct FoxgloveMsgsPoint3Mapper;

impl TypedTopicMapper for FoxgloveMsgsPoint3Mapper {
    type Ros = ros_env::foxglove_msgs::msg::Point3;
    type Bus = crate::foxglove_msgs::msg::v1::Point3;

    fn type_name(&self) -> &'static str {
        "foxglove_msgs/msg/Point3"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(point3_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(point3_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/foxglove_msgs/point3.py
"""Generated mapper for `foxglove_msgs/msg/Point3`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert


def point3_to_bus(msg):
    from robot_bus.foxglove_msgs.msg.v1 import Point3 as BusMsg

    bus = BusMsg()
    bus.x = msg.x
    bus.y = msg.y
    bus.z = msg.z
    return bus


def point3_to_ros(bus):
    from foxglove_msgs.msg import Point3 as RosMsg

    out = RosMsg()
    out.x = bus.x
    out.y = bus.y
    out.z = bus.z
    return out


class FoxgloveMsgsPoint3Mapper:
    def type_name(self) -> str:
        return "foxglove_msgs/msg/Point3"

    def ros_msg_type(self):
        from foxglove_msgs.msg import Point3 as RosMsg

        return RosMsg

    def ros_to_bus(self, msg) -> bytes:
        return point3_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        from robot_bus.foxglove_msgs.msg.v1 import Point3 as BusMsg

        bus = BusMsg()
        bus.ParseFromString(payload)
        return point3_to_ros(bus)
